src/core/server/saved_objects/export/collect_exported_objects.ts

```diff
@@ -126,3 +165,68 @@ const fetchReferences = async ({
       .map((obj) => ({ type: obj.type, id: obj.id })),
   };
 };
+
+const buildTransforms = (typeRegistry: ISavedObjectTypeRegistry) =>
+  typeRegistry.getAllTypes().reduce((transforms, type) => {
+    if (type.management?.onExport) {
+      return {
+        ...transforms,
+        [type.name]: type.management.onExport,
+      };
+    }
+    return transforms;
+  }, {} as Record<string, SavedObjectsExportTransform>);
+
+const buildIsExportable = (
+  typeRegistry: ISavedObjectTypeRegistry
+): SavedObjectsExportablePredicate<any> => {
+  const exportablePerType = typeRegistry.getAllTypes().reduce((transforms, type) => {
+    if (type.management?.isExportable) {
+      transforms[type.name] = type.management.isExportable;
+    }
+    return transforms;
+  }, {} as Record<string, SavedObjectsExportablePredicate>);
+
+  return (obj: SavedObject) => {
+    const typePredicate = exportablePerType[obj.type];
+    return typePredicate ? typePredicate(obj) : true;
```

Comment:
optional: this has a chance to fail if the saved object type collides with a property on the prototype. e.g. a saved object type of `toString` could resolve to the built-in function of the same name